mteb/abstasks/AbsTaskMultilabelClassification.py

```diff
@@ -0,0 +1,146 @@
+        if self.metadata_dict["main_score"] in scores:
+            scores["main_score"] = scores[self.metadata_dict["main_score"]]
+        else:
+            logger.warn(
+                f"main score {self.metadata_dict['main_score']} not found in scores {scores.keys()}"
+            )
+
+    def evaluate(self, model, eval_split="test", train_split="train", **kwargs):
+        if not self.data_loaded:
+            self.load_data()
+
+        if self.is_multilingual:
+            scores = {}
+            for lang in self.dataset:
+                logger.info(
+                    f"\nTask: {self.metadata_dict['name']}, split: {eval_split}, language: {lang}. Running..."
+                )
+                scores[lang] = self._evaluate_monolingual(
+                    model, self.dataset[lang], eval_split, train_split, **kwargs
+                )
+                self._add_main_score(scores[lang])
+        else:
+            logger.info(
+                f"\nTask: {self.metadata_dict['name']}, split: {eval_split}. Running..."
+            )
+            scores = self._evaluate_monolingual(
+                model, self.dataset, eval_split, train_split, **kwargs
+            )
+            self._add_main_score(scores)
+
+        return scores
+
+    def _evaluate_monolingual(
+        self, model, dataset, eval_split="test", train_split="train", **kwargs
+    ):
+        train_split = dataset[train_split]
+        eval_split = dataset[eval_split]
+        params = {"k": self.k, "batch_size": self.batch_size}
+        params.update(kwargs)
+
+        scores = []
+        test_cache, idxs = (
+            None,
+            None,
+        )  # we store idxs to make the shuffling reproducible
+        for i in range(self.n_experiments):
+            logger.info(
+                "=" * 10 + f" Experiment {i+1}/{self.n_experiments} " + "=" * 10
+            )
+            # Bootstrap `self.samples_per_label` samples per label for each split
+            X_sampled, y_sampled, idxs = self._undersample_data(
+                train_split["text"], train_split["label"], self.samples_per_label, idxs
+            )
+            binarizer = MultiLabelBinarizer()
+            y_train = binarizer.fit_transform(y_sampled)
+            y_test = binarizer.transform(eval_split["label"])
+            evaluator = kNNMultiLabelClassificationEvaluator(
+                X_sampled,
+                y_train,
+                eval_split["text"],
+                y_test,
+                **params,
+            )
+            scores_exp, test_cache = evaluator(model, test_cache=test_cache)
+            scores.append(scores_exp)
+
+        if self.n_experiments == 1:
+            return scores[0]
+        else:
+            avg_scores = {k: np.mean([s[k] for s in scores]) for k in scores[0].keys()}
+            std_errors = {
+                k + "_stderr": np.std([s[k] for s in scores]) for k in scores[0].keys()
+            }
+            return {**avg_scores, **std_errors}
```

Comment:
Hmm, we have previously discussed creating a standard format for results. This is a reasonable time to discuss it (at least the standard format for parts of it) as the inner results keys have multiple scores.

Let us ignore the format of lang, split nesting (that is for another PR)

Generally, all tasks have a singular main score. However for this case we either have to aggregate them, have multiple main scores or produce multiple results files from one task.

For me it seems the most reasonable to produce a list of results (one for each label), which then in a final table will be displayed as MyNewTask(cat1), MyNewTask(cat2)?

Reply:
Is this something I need to address in this PR, or should we postpone this for a separate issue of its own? If we decide to discuss it here, perhaps we should bring more people on board with this.